pkg/client/auth/kubernetes/token_credentials.go

```diff
@@ -0,0 +1,55 @@
+package kubernetes
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"golang.org/x/oauth2"
+)
+
+type NativeTokenCredentials struct {
+	TokenSource oauth2.TokenSource
+}
+
+func (c *NativeTokenCredentials) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	jwt, err := c.getJWT(c.TokenSource)
+	if err != nil {
+		return nil, err
+	}
+
+	ca, err := getClusterCA()
```

Comment:
are we doing this on every request?  Does it make any measurable difference to performance?

Reply:
I say this because we could obviously cache the encoded ca in memory, if only for a short time.

Reply:
I don't believe it will have a major difference on performance, I can look into caching it in memory though